substrate/client/basic-authorship/src/basic_authorship.rs

```diff
@@ -283,28 +283,48 @@ where
 		max_duration: time::Duration,
 		block_size_limit: Option<usize>,
 	) -> Self::Proposal {
-		let (tx, rx) = oneshot::channel();
-		let spawn_handle = self.spawn_handle.clone();
+		Self::propose(
+			self,
+			ProposeArgs {
+				inherent_data,
+				inherent_digests,
+				max_duration,
+				block_size_limit,
+				ignored_nodes_by_proof_recording: None,
+			},
+		)
+		.boxed()
```

Comment:
nit: I know it is a bit out of scope of this PR, but maybe `sp_consensus::Proposer` could be migrated to `async_trait` so we can get rid of future types?